search/reverse_geocoder.cpp

```diff
@@ -275,7 +275,7 @@ void ReverseGeocoder::GetNearbyBuildings(m2::PointD const & center, double radiu
   auto const addBuilding = [&](FeatureType & ft)
   {
     std::string const & hn = GetHouseNumber(ft);
-    if (hn.empty())
+    if (hn.empty() && !ft.HasStreetName())
```

Comment:
- HasStreetName is not about a street name actually.

- Since we work with _not ideal_ OSM, what should we show if we have the nearest (to the reverse geocoding point) building with street only and a bit far - a building with street and house number?

- Probably, I can fetch buildings with streets at this point, but not sure if it is worse or not ..

Reply:
>     * Since we work with _not ideal_ OSM, what should we show if we have the nearest (to the reverse geocoding point) building with street only and a bit far - a building with street and house number?

Not knowing how big is the difference between original OSM data and what OM has, I would say that the nearest building with street only is probably going to be the actual address of the POI. Who cares if there is a building nearby with street+housenumber if it's not the actual address of the POI?